backend/chainlit/server.py

```diff
@@ -870,13 +871,67 @@ async def upload_file(
     assert file.filename, "No filename for uploaded file"
     assert file.content_type, "No content type for uploaded file"
 
+    validate_file_upload(file)
+
     file_response = await session.persist_file(
         name=file.filename, content=content, mime=file.content_type
     )
 
     return JSONResponse(content=file_response)
 
 
+def validate_file_upload(file: UploadFile):
+    if config.features.spontaneous_file_upload is None:
+        return  # TODO: if it is not configured what should happen?
+
+    if config.features.spontaneous_file_upload.enabled is False:
+        raise HTTPException(
+            status_code=400,
+            detail="File upload is not enabled",
+        )
+
+    validate_file_mime_type(file)
+    validate_file_size(file)
+
+
+def validate_file_mime_type(file: UploadFile):
+    if config.features.spontaneous_file_upload.accept is None:
+        return
```

Comment:
Maybe add an explicit comment here that we'll allow any mime type unless a value is defined.